src/main/java/edu/indiana/dlib/amppd/service/impl/BundleServiceImpl.java

```diff
@@ -8,18 +13,54 @@
 import edu.indiana.dlib.amppd.model.Primaryfile;
 import edu.indiana.dlib.amppd.repository.BundleRepository;
 import edu.indiana.dlib.amppd.repository.PrimaryfileRepository;
+import edu.indiana.dlib.amppd.service.AmpUserService;
 import edu.indiana.dlib.amppd.service.BundleService;
 import lombok.extern.slf4j.Slf4j;
 
 @Service
 @Slf4j
 public class BundleServiceImpl implements BundleService {
-	
+
 	@Autowired
-    private BundleRepository bundleRepository;
-	
+	private BundleRepository bundleRepository;
+
 	@Autowired
-    private PrimaryfileRepository primaryfileRepository;
+	private PrimaryfileRepository primaryfileRepository;
+
+	@Autowired
+	private AmpUserService ampUserService;
+
+	/**
+	 * @see edu.indiana.dlib.amppd.service.BundleService.findAllNamed()
+	 */
+	public List<Bundle> findAllNamed() {
+		List<Bundle> bundles = bundleRepository.findAllWithNonEmptyNameNonEmptyPrimaryfiles();
+		log.info("Successfully found " + bundles.size() + " named bundles.");
+		return bundles;
+	}
+
+	/**
+	 * @see edu.indiana.dlib.amppd.service.BundleService.findNamedByCurrentUser(String)
+	 */
+	public Bundle findNamedByCurrentUser(String name) {
+		String username = ampUserService.getCurrentUsername();
+		List<Bundle> bundles = bundleRepository.findByNameAndCreatedBy(name, ampUserService.getCurrentUsername());
+
+		// bundle name shall be unique per user if all bundles are created via AMPPD UI;
+		// however just in case there're more than one found, we will return the first
+		if (bundles == null || bundles.isEmpty()) {
+			log.info("No bundle found with name " + name + " for the current user " + username);
+			return null;
+		}
+
+		if (bundles.size() > 1) {
+			log.warn("There are " + bundles.size() + " bundles found with name " + name + " for the current user " + username);
+		}
+
+		Bundle bundle = bundles.get(0);
+		log.info("Successfully found bundle " + bundle.getId() + " with name " + name + " for the current user " + username);
+		return bundle;
```

Comment:
See my other comment.  Wondering if this should return a list of bundles.

Reply:
in theory if bundle is only created via front end, then it should ensure that no duplicate exists. otherwise its a system error, not user error.

Reply:
I decided to throw exception if more than one bundle found, this ensures that the bundle updated is indeed what the user intended to, not some other bundle created outside of the UI.  We also need to clean up current table to make sure such case doesn't exist.